src/goto-programs/goto_check.cpp

```diff
@@ -231,6 +234,147 @@ void goto_checkt::overflow_check(
     guard);
 }
 
+void goto_checkt::input_overflow_check(
+  const expr2tc &expr,
+  const locationt &loc)
+{
+  code_function_call2t func_call = to_code_function_call2t(expr);
+  if(!is_symbol2t(func_call.function))
+    return;
+  const std::string func_name =
+    to_symbol2t(func_call.function).thename.as_string();
+
+  unsigned number_of_format_args, fmt_idx;
+
+  if(func_name.find("__ESBMC_scanf") != std::string::npos)
+  {
+    fmt_idx = 0;
+    number_of_format_args = func_call.operands.size() - 1;
+  }
+  else if(func_name.find("__ESBMC_fscanf") != std::string::npos)
+  {
+    fmt_idx = 1;
+    number_of_format_args = func_call.operands.size() - 2;
+  }
+  else
+    return;
+
+  // obtain the format string
+  const std::string fmt =
+    get_string_argument(func_call.operands[fmt_idx]).as_string();
+
+  // obtain the length limits in the format string
+  long unsigned int pos = 0;
+  std::vector<std::string> limits;
+
+  for(std::string tmp_str = ""; pos < fmt.length(); pos++)
```

Comment:
This can probably be generalized into its own class (e.g. printf). C uses format strings in many places. This could be left to another PR though